Fabric.Terminology.API/Scripts/Terminology-Intall-Utilities.psm1

```diff
@@ -1,28 +1,130 @@
+    }
+    else {
+        if ($Silent -eq $true) {
+            if ($Required -eq $true) {
+                Write-DosMessage -Level "Error"  -Message "$Prompt is required and was not provided through the command parameter nor the install.config." -ErrorAction Stop
+            }
+        }
+        elseif ($Required -eq $true) {
+            while ([string]::IsNullOrWhiteSpace($result)) {
+                $result = Read-Host "Enter the $Prompt $AdditionalPromptInfo".Trim()
+            }
+        }
+        else {
+            $result = Read-Host "Enter the $Prompt $AdditionalPromptInfo".Trim()
+        }
+
+        return $result
+    }
+}
+
+function Invoke-SqlCommand{
+    param(
+        [String] $SqlServerAddress,
+        [String] $DatabaseName,
+        [String] $Query,
+        [PSCustomObject] $Parameters= @{}
+    )
+
+    $connectionString = "Data Source=$SqlServerAddress;Initial Catalog=$($DatabaseName); Trusted_Connection=True;"
+    $connection = New-Object System.Data.SqlClient.SQLConnection($connectionString)
+    $command = New-Object System.Data.SqlClient.SqlCommand($Query, $connection)
+    
+    try {
+        foreach($p in $Parameters.Keys){		
+          $command.Parameters.AddWithValue("@$p",$Parameters[$p]) | Out-Null
+         }
+
+        $connection.Open() 
+        $command.ExecuteNonQuery() | Out-Null
+        $connection.Close()        
+    }catch [System.Data.SqlClient.SqlException] {
+        Write-DosMessage -Level "Error" -Message "An error ocurred while executing the command. 
+        Connection String: $($connectionString)"
+        throw
+    }
 }
 
 function Get-TerminologyConfig {
     param(
-        [PSCredential] $Credentials
+        [PSCredential] $Credentials,
+        [String] $DiscoveryServiceUrl,
+        [String] $SqlAddress,
+        [String] $MetadataDbName,
+        [String] $AppInsightsKey,
+        [String] $IisUserName,
+        [SecureString] $IisUserPassword,
+        [String] $AppName,
+        [switch] $Silent
     )
 
-    $config = [TerminologyConfig]::new()        
+    # Get Configuration
+    $installSettings = Get-InstallationSettings "terminology"
+    Write-Host $installSettings
 
     # Get Credentials
-    If ($Null -ne $Credentials) {
-        $config.iisUserCredentials = $Credentials
+    if ($Null -ne $Credentials) {
+        $iisUserCredentials = $Credentials
     }
     else {
-        $iisUser = Read-Host "Please enter the IIS App Pool User"
-        $userEnteredPassword = Read-Host "Enter the password for $iisUser" -AsSecureString
-        $credential = New-Object -TypeName "System.Management.Automation.PSCredential" -ArgumentList $iisUser, $userEnteredPassword
+        $iisUserConfig = Get-ConfigValue -Prompt "IIS user name to run the app pool" -DefaultFromParam $IisUserName -DefaultFromInstallConfig $installSettings.iisUser -Silent $Silent
+        
+        $passwordPrompt = "IIS user password"
```

Comment:
To me "App pool user password" would be better than "IIS user password"

Reply:
I like it